scripts/gha/test_lab.py

```diff
@@ -212,6 +206,36 @@ def main(argv):
       extra_info=" (ON REAL DEVICE VIA FTL)")
 
 
+def _run_test_on_ftl(tests, tested_tests, retry=3):
+  threads = []
+  for test in tests:
+    logging.info("Start running testapp: %s" % test.testapp_path)
+    thread = threading.Thread(target=test.run)
+    threads.append(thread)
+    thread.start()
+    tested_tests.append(test)
+  for thread in threads:
+    thread.join()
+
+  if retry > 1:
+    failed_tests = []
+    for test in tests:
+      result = test_validation.validate_results(test.logs, test_validation.CPP)
+      if not result.complete or result.fails != 0:
+        logging.info("Failure(s) occurred, testapp: %s" % test.testapp_path)
+        failed_test = Test(
+              device=test.device,
+              platform=test.platform,
+              testapp_path=test.testapp_path,
+              results_dir=test.results_dir + "retry")
+        failed_tests.append(failed_test)
+    if failed_tests:
+      logging.info("Failure(s) occurred, retry test(s) on FTL.")
```

Comment:
`retrying` instead of `retry`?  

`retry` makes me feel like I need to take action.
`retrying` makes me feel like someone is already taking action for me.